gtp_wrapper.py

```diff
@@ -12,147 +12,32 @@
-        if self.should_resign():
-            return gtp.RESIGN
-        return coords.to_pygtp(move)
-
-    def final_score(self):
-        return self.position.result_string()
-
-    def showboard(self):
-        print('\n\n' + str(self.position) + '\n\n', file=sys.stderr)
-        return True
-
-    def should_resign(self):
-        raise NotImplementedError
-
-    def get_score(self):
-        return self.position.result_string()
-
-    def suggest_move(self, position):
-        raise NotImplementedError
-
-    def play_move(self, c):
-        raise NotImplementedError
-
-    def initialize_game(self):
-        raise NotImplementedError
-
-    def chat(self, msg_type, sender, text):
-        raise NotImplementedError
-
-    def to_sgf(self):
-        raise NotImplementedError
-
-
-class MCTSPlayer(MCTSPlayerMixin, GtpInterface):
-    pass
-
-
-class CGOSPlayer(CGOSPlayerMixin, GtpInterface):
-    pass
-
-
-class KGSPlayer(MCTSPlayer):
-    """ A wrapper for playing on KGS
-    Adds 'courtsey pass' -- if the opponent passes, we pass, regardless
-    of score or our opinion on the game. """
-
-    def __init__(self, **kwargs):
-        self.they_passed = False
-        super().__init__(**kwargs)
-
-    def get_move(self, color):
-        if self.they_passed:
-            return gtp.PASS
-        return super().get_move(color)
-
-    def make_move(self, color, vertex):
-        if vertex == gtp.PASS:
-            self.they_passed = True
-        else:
-            self.they_passed = False
-        return super().make_move(color, vertex)
-
-    def clear(self):
-        self.they_passed = False
-        return super().clear()
+from strategies import MCTSPlayer, CGOSPlayer
 
 
 def make_gtp_instance(read_file, readouts_per_move=100, verbosity=1, cgos_mode=False, kgs_mode=False):
     n = DualNetwork(read_file)
     if cgos_mode:
-        instance = CGOSPlayer(n, seconds_per_move=5, timed_match=True,
-                              verbosity=verbosity, two_player_mode=True)
-    elif kgs_mode:
-        instance = KGSPlayer(network=n, num_readouts=readouts_per_move,
-                             verbosity=verbosity, two_player_mode=True)
+        player = CGOSPlayer(network=n, seconds_per_move=5, timed_match=True,
+                            verbosity=verbosity, two_player_mode=True)
     else:
-        instance = MCTSPlayer(network=n, num_readouts=readouts_per_move,
-                              verbosity=verbosity, two_player_mode=True)
+        player = MCTSPlayer(network=n, num_readouts=readouts_per_move,
+                            verbosity=verbosity, two_player_mode=True)
+
     name = "Minigo-" + os.path.basename(read_file)
-    gtp_engine = gtp_extensions.GTPDeluxe(instance, name=name)
-    return gtp_engine
+    version = "0.2"
+
+    engine = gtp_engine.Engine()
+    engine.add_cmd_handler(
+        gtp_engine.EngineCmdHandler(engine, name, version))
+
+    engine.add_cmd_handler(KgsCmdHandler(player))
```

Comment:
Is it a good idea to register these handlers in the `if cgos_mode` blocks, so that we don't accidentally handle commands we didn't intend to handle in certain modes?

Reply:
done.